app/assets/v2/js/activity.js

```diff
@@ -303,53 +303,95 @@ $(document).ready(function() {
       return;
     }
 
-    // user input
-    var comment = $parent.parents('.box').find('.comment_container textarea').val();
+    if ($('.editableComment')[0]) {
+      // edited content
+      let comment = $parent.parents('.box').find('.editableComment').val();
 
     $parent.parents('.box').find('.comment_container textarea').prop('disabled', true);
-    $('.post_comment').prop('disabled', true);
+    $('.editableComment').prop('disabled', true);
 
-    // validation
-    if (!comment) {
-      return;
-    }
-
-    $parent.parents('.activity.box').find('.loading').removeClass('hidden');
-    var has_hidden_comments = $parent.parents('.activity.box').find('.row.comment_row.hidden').length;
-    // increment number
-    var num = $parent.find('span.num').html();
+      // validation
+      if (!comment) {
+        $('.editableComment').prop('disabled', false);
+        return;
+      }
 
-    num = parseInt(num) + 1;
-    $parent.find('span.num').html(num);
+      $parent.parents('.activity.box').find('.loading').removeClass('hidden');
 
-    // remote post
-    var params = {
-      'method': 'comment',
-      'comment': comment,
-      'csrfmiddlewaretoken': $('input[name=csrfmiddlewaretoken]').val()
-    };
-    var url = '/api/v0.1/activity/' + $parent.data('pk');
+      let comment_id = $('.editableComment').parent().closest('.comment_row').data('id');
+      let url = '/api/v0.1/comment/' + comment_id;
+      let params = {
+        'method': 'EDIT',
+        'comment': comment,
+        'csrfmiddlewaretoken': $('input[name=csrfmiddlewaretoken]').val()
+      };
 
-    $.post(url, params, function(response) {
-      var success_callback = function($parent) {
-        $parent.parents('.box').find('.comment_container textarea').val('');
-        $parent.find('textarea').focus();
+      $.post(url, params, function(response) {
+        let success_callback = function($parent) {
+          // $parent.find('textarea').focus();
+        };
```

Comment:
should we remove this if it's not doing anything?

Reply:
The commented line or the entire `success_callback` block @thelostone-mc?

Reply:
replaced commented out code with `$.noop(); // do nothing`